src/nvim/buffer_defs.h

```diff
@@ -664,12 +664,13 @@ struct file_buffer {
   char_u      *b_p_efm;         /* 'errorformat' local value */
   char_u      *b_p_ep;          /* 'equalprg' local value */
   char_u      *b_p_path;        /* 'path' local value */
-  int b_p_ar;                   /* 'autoread' local value */
+  bool b_p_ar;                  // 'autoread' local value, when present
+  bool has_b_p_ar;              // whether buffer has an 'autoread' local value
```

Comment:
ok, I suppose `b_p_ar` could be renamed here, to indicate the representation is not the same as in vim. Or if you prefer I could try to fix the bug still using the old representation. But no matter how, as this is the first global-local boolean option, we are _creating_ the precedent here.